### PR TITLE
feat(api-markdown-documenter): Allow configuration of "alerts" in child item tables

### PR DESCRIPTION
**Pull request by Josmithr**
This PR...
* changes default alert text from `UPPERCASE` to standard sentence casing to be less jarring and play nicer with default line-wrapping policies.
    * See update test collateral for examples.
* Adds `getSingleLineExcerptText` to the `ApiItemUtilities` namespace export. This is useful if anyone wishes to override the default heading text policy (which the fluidframework.com build will leverage). 

[AB#8406](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/8406)

### CODE REVIEW

**Comment by Josmithr** [comment]
These changes are required to fix some misleading information in the fluidframework.com documentation. See draft PR leveraging these changes here: #21525

File: tools/api-markdown-documenter/src/api-item-transforms/helpers/TableHelpers.ts [hunk 1]
```diff
@@ -221,8 +205,9 @@ export function createDefaultSummaryTable(
 		return undefined;
 	}
 
-	// Only display "Alerts" column if there are any deprecated or alpha/beta items in the list.
-	const hasAlerts = doItemsContainAlerts(apiItems);
+	// Only display "Alerts" column if there are any alerts to display.
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/api-item-transforms/helpers/TableHelpers.ts:208]: The volume of changes required in this file have convinced me that a unified table builder function is warranted here. But I will do that in a future PR.

File: tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts [hunk 2]
```diff
@@ -465,3 +466,19 @@ export function getSafeFilenameForName(apiItemName: string): string {
 	// https://github.com/microsoft/rushstack/issues/1308
 	return apiItemName.replace(badFilenameCharsRegExp, "_").toLowerCase();
 }
+
+/**
+ * Extracts the text from the provided excerpt and adjusts it to be on a single line, and to omit any trailing `;`.
+ *
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts:472]:
```suggestion
 * @remarks Useful when a shortened version of a code excerpt is wanted, and especially if the contents need to fit on a single line.
```

**Review by alexvy86 — APPROVED** [review]
[Small nit in the other PR](https://github.com/microsoft/FluidFramework/pull/21525#discussion_r1646725594), otherwise LGTM.

**Review by DLehenbauer — APPROVED** [review]

**Pull request merged**